.github/workflows/soak.yml

```diff
@@ -245,40 +265,17 @@ jobs:
 
       - name: Analyze captures
         run: |
-          ./soaks/bin/analyze_experiment.sh ${{ github.event.number }}-captures/ ${{ needs.compute-soak-meta.outputs.baseline-sha }} ${{ needs.compute-soak-meta.outputs.comparison-sha }}
-
-  observer:
-    name: Build and push 'observer' to Github CR
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Check out the repo
-        uses: actions/checkout@v2.4.0
-
-      - name: Set up Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v1
-
-      - name: Log in to the Container registry
-        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          ./soaks/bin/analyze_experiment.sh ${{ github.event.number }}-captures/ ${{ needs.compute-soak-meta.outputs.baseline-sha }} ${{ needs.compute-soak-meta.outputs.comparison-sha }} > /tmp/${{ github.event.number}}-${{ github.run_number }}-analysis
 
-      - name: Extract metadata (tags, labels) for Docker
-        id: meta
-        uses: docker/metadata-action@548e2346a9987b56d8a4104fe776321ff8e23440
+      - name: Read analysis file
+        id: read-analysis
+        uses: juliangruber/read-file-action@v1
         with:
-          images: ghcr.io/${{ github.repository }}/soak-observer
-          tags: type=sha, format=long
+          path: /tmp/${{ github.event.number }}-${{ github.run_number }}-analysis
 
-      - name: Build and push Docker image
-        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+      - name: Post Results To PR
+        uses: peter-evans/create-or-update-comment@v1
         with:
-          context: .
-          file: lib/soak/Dockerfile
-          push: true
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
-          cache-from: type=gha, scope=${{ github.workflow }}
-          cache-to: type=gha, scope=${{ github.workflow }}
+          issue-number: ${{ needs.compute-soak-meta.outputs.pr-number }}
+          edit-mode: replace
+          body: ${{ steps.read-analysis.outputs.content }}
```

Comment:
These actions seam simple enough that I think I would have preferred to just use a script action than pull in extra dependencies but it is not a huge deal.

Reply:
You would think, but I can't write javascript. I'm open to PRs that remove the need for these.